src/giskardpy/python_interface/python_interface.py

```diff
@@ -1856,3 +2116,54 @@ def get_result(self, timeout: rospy.Duration = rospy.Duration()) -> MoveResult:
 
     def _feedback_cb(self, msg: MoveFeedback):
         self.last_feedback = msg
+
+    def get_end_motion_reason(self, move_result: MoveResult = None, show_all=False):
+        if not move_result and not self.last_execution_state:
+            raise Exception('No MoveResult available to analyze')
+        elif not move_result:
+            execution_state = self.last_execution_state
+        else:
+            execution_state = move_result.execution_state
+
+        result = {}
+
+        endMotion_idx = 0
+        cancelMotion_idx = 0
+        idx = 0
+        for monitor in execution_state.monitors:
+            if monitor.monitor_class == 'CancelMotion':
```

Comment:
`CancelMotion.__name__`

Reply:
there might be multiple end/cancel monitors

Reply:
there is also a bug/problem that i have at other places too. monitors which inherit from cancel monitor don't evaluate to true here

Reply:
cancelMotion_idx is not used

Reply:
ever heard of enumerate?